From fce2e96b9d9fd827ece32d88a88aa988123a0074 Mon Sep 17 00:00:00 2001
From: Lukas Tribus <luky-37@hotmail.com>
Date: Tue, 12 Feb 2013 22:13:19 +0100
Subject: DOC: simplify bind option "interface" explanation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The current documentation of the bind option "interface" can be misleading
(as seen on the ML recently).

This patch tries to address misunderstandings by :

  - avoiding the words listen or bind in the behavior description, using
    "restrict to interface" instead

  - using a different sentence construction (partially stolen from
    "man 7 socket": SO_BINDTODEVICE)

  - "defragmentation": moving behavior related explanations to the beginning
    and restrictions, use-cases and requirements to the end.
---
 doc/configuration.txt |   15 +++++++--------
 1 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/doc/configuration.txt b/doc/configuration.txt
index 30647a9..5f7a65b 100644
--- a/doc/configuration.txt
+++ b/doc/configuration.txt
@@ -7126,14 +7126,13 @@ id <id>
   option can only be used when defining only a single socket.
 
 interface <interface>
-  Sets the name of the network interface to listen. This is currently only
-  supported on Linux. The interface must be a primary system interface, not an
-  aliased interface. When specified, all addresses on the same line will only
-  be accepted if the incoming packets physically come through the designated
-  interface. It is also possible to bind multiple frontends to the same address
-  if they are bound to different interfaces.  Note that binding to a network
-  interface requires root privileges. This parameter is only compatible with
-  TCPv4/TCPv6 sockets.
+  Restricts the socket to a specific interface. When specified, only packets
+  received from that particular interface are processed by the socket. This is
+  currently only supported on Linux. The interface must be a primary system
+  interface, not an aliased interface. It is also possible to bind multiple
+  frontends to the same address if they are bound to different interfaces. Note
+  that binding to a network interface requires root privileges. This parameter
+  is only compatible with TCPv4/TCPv6 sockets.
 
 level <level>
   This setting is used with the stats sockets only to restrict the nature of
-- 
1.7.1

